core/src/main/java/org/elasticsearch/action/support/master/AcknowledgedResponse.java

```diff
@@ -61,4 +68,41 @@ protected void readAcknowledged(StreamInput in) throws IOException {
     protected void writeAcknowledged(StreamOutput out) throws IOException {
         out.writeBoolean(acknowledged);
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field("acknowledged", acknowledged);
+        builder.endObject();
+        return builder;
+    }
+
+    protected static void parseInnerToXContent(XContentParser parser, AcknowledgedResponse.Builder context) throws IOException {
+        XContentParser.Token token = parser.currentToken();
```

Comment:
given this hierarchy, maybe we should check already if we support classes that add their custom fields to the response, like `CreateIndexResponse`. Maybe we can add parsing code for that one too (only parsing, without adding the rest of the API)? What do you think?